x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java

```diff
@@ -136,6 +143,13 @@ static IndexLifecycleExplainResponse getIndexLifecycleExplainResponse(
                 )
             ) {
                 phaseExecutionInfo = PhaseExecutionInfo.parse(parser, currentPhase);
+
+                // Try to add default rollover conditions to the response.
+                var rolloverAction = (RolloverAction) phaseExecutionInfo.getPhase().getActions().get(RolloverAction.NAME);
+                if (rolloverAction != null) {
+                    var conditions = applyDefaultConditions(rolloverAction.getConditions(), rolloverOnlyIfHasDocuments);
+                    phaseExecutionInfo.getPhase().getActions().put(RolloverAction.NAME, new RolloverAction(conditions));
+                }
```

Comment:
Any suggestions on different approaches where we don't have to inject the default conditions like this in response are welcome.

Reply:
This looks good to me. I think the only thing that would be nice here is if we'd be able to indicate in the output that these conditions are implicit / automatically configured as opposed to user specified. 

That would increase the complexity here quite a bit (extending `Condition` to know when it's implicit and when it's explicitly configured and changing the xcontent representation accordingly - which might have implications in the UI if an accidental implicit representation reaches the cluster state for an ILM policy definition). 
I think displaying the extra conditions without an "implicit" indication (as proposed now) already adds a lot of value for users to understand exactly what rollover is ILM executing. 

@dakrone would you agree here?

Reply:
Yeah I thought about that as well. My reasoning was also that it would require quite a bit of work to make a distinction between explicit and implicit conditions in the API response. Either we would have to do what you said (changing the structure of the underlying conditions/classes), or we would have to create a dedicated response class for the rollover conditions in the explain API (instead of using the `RolloverConditions` class that is also used to serialize/cache the lifecycle state in the cluster metadata). I think the second option might be doable, but I wasn't sure whether it'd be worth the effort.

Reply:
++ since we're documenting the implicit conditions I'd say we go ahead with this as is